<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
  <script>
    /**1.只要有异常对象抛出（不管是浏览器抛出还是代码主动抛出）都会让程序停止执行，如果想让
     *      程序继续执行或被try...catch捕获
     *2.每个错误类型都可以传入一个参数，表示实际的错误信息
     * */

    /**自定义异常*/
    function MyError(message){
      this.message=message
      this.name='自定义错误'
    }
    //自定义错误类型继承基本错误类型
    MyError.prototype=new Error()
    try{
      throw new MyError("注意：这是自定义类型错误")
    }catch(error){
      console.log(error.message)
    }
  </script>
</body>
</html>